forte/processors/misc/remote_processor.py

```diff
@@ -56,16 +57,73 @@ class RemoteProcessor(PackProcessor):
-                "Remote service not started or invalid endpoint configs."
+                f"{response.status_code} {response.reason}: Please double "
+                "check your endpoint URL configuration and make sure that the "
+                f"remote service at {self.configs.url} is a valid pipeline "
+                "service that is up and running."
+            )
+        service_name: str = response.json()["service_name"]
+        input_format: str = response.json()["input_format"]
+
+        if self._validation.do_init_type_check:
+            # Validate service name and input format
+            if service_name != self._validation.expected_name:
+                raise ProcessorConfigError(
+                    "Validation fail: The expected service name "
+                    f"('{self._validation.expected_name}') does not match "
+                    "the actual name returned by remote service "
+                    f"('{service_name}'). Consider updating the configs of "
+                    "RemoteProcessor so that 'validation.expected_name' "
+                    f"equals to '{service_name}'."
+                )
+            if input_format != self._validation.input_format:
+                raise ProcessorConfigError(
+                    "Validation fail: The expected input format "
+                    f"('{self._validation.input_format}') does not match "
+                    "the actual input format returned by remote service "
+                    f"('{input_format}'). Consider updating the configs of "
+                    "RemoteProcessor so that 'validation.input_format' "
+                    f"equals to '{input_format}'."
+                )
+
+            # Get the output records
+            response = self._requests.get(f"{self.configs.url}/records")
+            if response.status_code != 200 or response.json()["status"] != "OK":
+                raise ProcessorConfigError(
+                    f"{response.status_code} {response.reason}: "
+                    "Fail to fetch records from remote service. Please make "
+                    f"sure that the remote service at {self.configs.url} is "
+                    "a valid pipeline service that is up and running."
+                )
+            self._records = response.json()["records"]
+
+    def check_record(self, input_pack):
+        r"""Add additional checking of remote pipeline service. The records
+        of remote service will be queried and validated against the expected
+        record types and attributes in configs.
+
+        Args:
+            input_pack: The input datapack.
+        """
+        super().check_record(input_pack)
+        if (
+            self._validation.do_init_type_check
+            and self._validation.expected_records is not None
+        ):
+            # Validate the output records
+            record_types_and_attributes_check(
+                self._validation.expected_records.todict(), self._records
             )
```

Comment:
`do_init_type_check` should be moved to the base too (along with the configurations)